test/PrintAsCxx/lit.local.cfg

```diff
@@ -0,0 +1,12 @@
+# Make a local copy of the substitutions.
+config.substitutions = list(config.substitutions)
+
+config.substitutions.insert(0, ('%check-in-clang\+\+',
+  '%%clang --driver-mode=g++ -fsyntax-only -x c++-header '
```

Comment:
Also, I'd kind of like to understand what all these options are and why they're needed.

Reply:
I'll add some comments in a follow-up PR.